From 01da297611243be689d636785172ee27fc9adc8d Mon Sep 17 00:00:00 2001
From: Wang Panzhenzhuan <randy.wang@rock-chips.com>
Date: Tue, 31 Mar 2020 18:08:13 +0800
Subject: [PATCH] media: rockchip: cif: delete mipi dphy subdev s_stream in
 csi2_start/stop

mipi dphy subdev s_stream has been called in rkcif_pipeline_set_stream,
no need be called here

Signed-off-by: Wang Panzhenzhuan <randy.wang@rock-chips.com>
Change-Id: Iba9074da59d6516b169660ebae8039b70ee428c4
---
 drivers/media/platform/rockchip/cif/mipi-csi2.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/media/platform/rockchip/cif/mipi-csi2.c b/drivers/media/platform/rockchip/cif/mipi-csi2.c
index f0fd1a5e5f02..0a24b73e9a2d 100644
--- a/drivers/media/platform/rockchip/cif/mipi-csi2.c
+++ b/drivers/media/platform/rockchip/cif/mipi-csi2.c
@@ -202,24 +202,12 @@ static int csi2_start(struct csi2_dev *csi2)
 	csi2_enable(csi2, host_type);
 
 	pr_debug("stream sd: %s\n", csi2->src_sd->name);
-	ret = v4l2_subdev_call(csi2->src_sd, video, s_stream, 1);
-	ret = (ret && ret != -ENOIOCTLCMD) ? ret : 0;
-	if (ret)
-		goto err_assert_reset;
 
 	return 0;
-
-err_assert_reset:
-	csi2_disable(csi2);
-	clk_disable_unprepare(csi2->pix_clk);
-	return ret;
 }
 
 static void csi2_stop(struct csi2_dev *csi2)
 {
-	/* stop upstream */
-	v4l2_subdev_call(csi2->src_sd, video, s_stream, 0);
-
 	csi2_disable(csi2);
 	clk_disable_unprepare(csi2->pix_clk);
 }
-- 
2.35.3

